### PR TITLE
Feature: Post and Reply by Threads

### PR DESCRIPTION
**Pull request by thai-d-v**
@Dimillian Sorry, I stated this PR before #1735 opened, and I just read it after I submitted this PR. I noticed that you have already self-assigned. In case my work does not align with your plan, I will close it.

This feature allows users to post and even reply to a status by a thread.
- all statuses are fully functional individually
- all statuses have the same visibility
- the maximum number of follow-up statuses is 5, be cautious when increasing it
- already localized

https://github.com/Dimillian/IceCubesApp/assets/46838577/def71c4b-c941-4b32-b7b8-a7550c73497b

### CODE REVIEW

**Comment by Dimillian** [comment]
Impressive!

File: Packages/Status/Sources/Status/Editor/StatusEditorView.swift [hunk 1]
```diff
@@ -14,230 +14,350 @@ import UIKit
...
+  @State private var mainSEVM: StatusEditorViewModel
+  @State private var followUpSEVMs: [StatusEditorViewModel] = []
+  @FocusState private var isSpoilerTextFocused: UUID? // connect CoreEditor and StatusEditorAccessoryView
+  @State private var editingMediaContainer: StatusEditorMediaContainer?
+  @State private var scrollID: UUID?
 
-  @State private var isDismissAlertPresented: Bool = false
-  @State private var isLanguageConfirmPresented = false
+  @FocusState private var editorFocusState: EditorFocusState?
+  private var focusedSEVM: StatusEditorViewModel {
```
Comment (Dimillian) [inline Packages/Status/Sources/Status/Editor/StatusEditorView.swift:27]: This is really nice!

**Comment by thai-d-v** [comment]
@Dimillian I am facing issues with localized strings. Normally, when I add a new code with `LocalizedStringKey`, Xcode automatically adds that key to the string catalog file. However, lately, Xcode is not doing it for me. Do you have any idea why?

**Comment by Dimillian** [comment]
Same and no, all the settings are on so I don't know why. 
Also usually if there is a missing translation iOS is supposed to use the en base, but it's not, so I need to add the default English to every language.

**Comment by Dimillian** [comment]
Just renaming a few stuff and moving around files. I think it's good! 
I'm wondering if we really need a main VM and the array, could we just have an array of ViewModel with 0 being the main?

**Comment by thai-d-v** [comment]
I intend to restructure the UI to make it easier for me to understand the code when I come back to it later. This will also make the logic code safer and easier to write. For instance, with this new code, it will be less likely to accidentally remove the main core editor.

https://github.com/thai-d-v/IceCubesApp/blob/4efe32c53665ddaf00bec29daf78cfadcfe7e4df/Packages/Status/Sources/Status/Editor/StatusEditorCoreView.swift#L97-L105

**Comment by singingwolfboy** [comment]
> the maximum number of follow-up statuses is 5, be cautious when increasing it

Out of curiosity, why is there a limit at all? And what should we be cautious about when increasing the limit?

**Comment by thai-d-v** [comment]
@singingwolfboy To prevent this feature from becoming a spam machine.

**Pull request merged**